ckanext/qgov/common/authenticator.py

```diff
@@ -39,32 +48,30 @@ def authenticate(self, environ, identity):
         """
         if 'login' not in identity or 'password' not in identity:
             return None
-        user = User.by_name(identity.get('login'))
+        login_name = identity.get('login')
+        user = User.by_name(login_name)
         if user is None:
-            LOG.debug('Login failed - username %r not found', identity.get('login'))
+            LOG.debug('Login failed - username %r not found', login_name)
             return None
 
-        qgov_user = Session.query(QGOVUser).filter_by(name=identity.get('login')).first()
-        if qgov_user.login_attempts >= 10:
-            LOG.debug('Login as %r failed - account is locked', identity.get('login'))
+        cache_key = '{}.ckanext.qgov.login_attempts.{}'.format(g.site_id, login_name)
```

Comment:
are the using redis for login attempts instead of db? that is very nice.

Reply:
2.9 DB modification needed a different SQLAlchemy interaction. I decided to instead take a leaf out of ckanext-security's book and use Redis.